engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -155,12 +155,12 @@ private[spel] class Typer(classLoader: ClassLoader, commonSupertypeFinder: Commo
           }
         }
 
-      case e: MethodReference => typeChildren(validationContext, node, current.popStack) { typedParams =>
-        TypeMethodReference(e.getName, current.stack, typedParams) match {
-          case Right(typingResult) => Valid(typingResult)
-          case Left(errorMsg) => if(strictMethodsChecking) invalid(errorMsg) else Valid(Unknown)
+      case e: MethodReference =>
+        if (current.isEmptyStack) {
```

Comment:
maybe let's remove popStack and stackTail altogether? And force to use stackHead with proper pattern matching? 
Now in `extractMethodReference` you assume it will be invoked after checking the stack is not empty - which is a bit error prone?

Reply:
Ugh.. As awlays, you're right :)